ArrCharOps.java

```diff
@@ -126,7 +181,28 @@ public static long hashCode(char[] arr) {
      *         return -2 if there is an error with the input.
      */
     public static int compareTo(String str1, String str2) {
-        // Replace the following statement with your code
-        return 0;
+        if (str1.isEmpty() || str2.isEmpty()) return -2;
+        int len1 = str1.length();
+      int len2 = str2.length();
+      int len3 = 0;
+      if(len1!=len2) len3=Math.min(len1, len2);
+      else len3= len1;
```

Comment:
1. use trenary op 
`len3 = len1 != len2 ? Math.min(len1, len2) : len1;`
but think about it for a second... 
let `n` be some int value. `Math.min(n, n) = n`, hence: `len3 = Math.min(len1, len2)` would be more than enough..